v2.4.1 build error

**Relevant system information:**
 - OS: CentOs 8
 - PostgreSQL version (output of `postgres --version`): 13.4
 - TimescaleDB version (output of `\dx` in `psql`): 2.4.1
 - Installation method: source

**Describe the bug**
`bootstrap` produces error, see below.

**To Reproduce**
```
$ ./bootstrap -DREGRESS_CHECKS=OFF -DWARNINGS_AS_ERRORS=OFF

-- The C compiler identification is GNU 8.4.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- TimescaleDB version 2.4.1. Can be updated from version 2.4.0
-- Build type is Release
-- Install method is 'source'
-- Performing Test CC_SUPPORTS_NO_UNUSED_CLI_ARG
-- Performing Test CC_SUPPORTS_NO_UNUSED_CLI_ARG - Success
-- Performing Test CC_SUPPORTS_NO_FORMAT_TRUNCATION
-- Performing Test CC_SUPPORTS_NO_FORMAT_TRUNCATION - Success
-- Performing Test CC_STRINGOP_TRUNCATION
-- Performing Test CC_STRINGOP_TRUNCATION - Success
-- Performing Test CC_SUPPORTS_IMPLICIT_FALLTHROUGH
-- Performing Test CC_SUPPORTS_IMPLICIT_FALLTHROUGH - Success
-- Performing Test CC_SUPPORTS_VISIBILITY_HIDDEN
-- Performing Test CC_SUPPORTS_VISIBILITY_HIDDEN - Success
-- Using compiler GNU
-- Found Git: /usr/bin/git (found version "2.27.0")
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
-- Unable to get git commit information
-- Compiling against PostgreSQL version 13.4
CMake Warning at CMakeLists.txt:423 (message):
  clang-format version 7 or 8 required


-- Regress checks and isolation checks disabled
-- Assertion checks are OFF
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1g")
-- Using OpenSSL version 1.1.1g
-- Found Perl: /usr/bin/perl (found suitable version "5.26.3", minimum required is "5.8")
-- Using nm /usr/bin/nm
-- Not generating downgrade script: downgrade generation disabled.
CMake Error at tsl/test/CMakeLists.txt:125 (add_dependencies):
  Cannot add target-level dependencies to non-existent target "installcheck".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.


-- Configuring incomplete, errors occurred!
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -73,24 +73,22 @@ elseif(REQUIRE_ALL_TESTS)
       "All tests were required but 'pg_isolation_regress' could not be found")
 endif()
 
-if(_install_checks)
-  add_custom_target(installcheck DEPENDS ${_install_checks})
+# We add the installcheck target even when _install_checks is empty as tsl code
+# might add dependencies to it even when regress checks are disabled.
+add_custom_target(installcheck DEPENDS ${_install_checks})
 
-  # Define a post test hook that is invoked after the installcheck target
-  # finishes. One can use add_dependencies on post hook target to run other
-  # targets after tests complete. This is used, e.g., by code coverage.
-  add_custom_target(installcheck-post-hook COMMENT "Post test hook")
-  add_custom_command(
-    TARGET installcheck
-    POST_BUILD
-    COMMAND cmake --build ${CMAKE_CURRENT_BINARY_DIR} --target
-            installcheck-post-hook)
-endif()
+# Define a post test hook that is invoked after the installcheck target
+# finishes. One can use add_dependencies on post hook target to run other
+# targets after tests complete. This is used, e.g., by code coverage.
+add_custom_target(installcheck-post-hook COMMENT "Post test hook")
+add_custom_command(
+  TARGET installcheck
+  POST_BUILD
+  COMMAND cmake --build ${CMAKE_CURRENT_BINARY_DIR} --target
+          installcheck-post-hook)
 
 # installchecklocal tests against an existing postgres instance
-if(_local_install_checks)
-  add_custom_target(installchecklocal DEPENDS ${_local_install_checks})
-endif()
+add_custom_target(installchecklocal DEPENDS ${_local_install_checks})
 
 add_subdirectory(sql)
 add_subdirectory(isolation)

